electron/launcher/src/_reducers/patcher.reducer.js

```diff
@@ -11,17 +11,28 @@ const INITIAL_STATE = {
     retryTime: 0,
     error: null
   },
-  isLatest: false
+  playButtonLock: false,
+  reqInitialCheck: true,
+  gameLaunchQueued: false,
+  patching: false
 }
 
 export default (state = INITIAL_STATE, action) => {
   switch(action.type) {
+    case types.CHECKING_FOR_UPDATE:
+      return { ...state, playButtonLock: true}
+    case types.INITIAL_CHECK_COMPLETE:
+      return { ...state, reqInitialCheck: false}
+    case types.QUEUE_GAME_LAUNCH:
+      return { ...state, gameLaunchQueued: true}
     case types.IS_LATEST_VERSION:
-      return { ...state, isLatest: true };
-    case types.UPDATE_AVAILABLE:
-      return { ...state, isLatest: false };
+      return { ...state, playButtonLock: false, patching: false};
```

Comment:
you are guaranteed that if you arent patching and play button lock is false, then latest version is true?

Reply:
No, if someone leaves the launcher on for a long time they may eventually become out of date. However, we always check for update before launching the game. In `Header.js` where we launch the game, you can see we launch the game by checking for update but telling the update function to queue an action (launch game).